core/amber/src/main/scala/edu/uci/ics/amber/engine/common/promise/PromiseBody.scala

```diff
@@ -0,0 +1,5 @@
+package edu.uci.ics.amber.engine.common.promise
+
+// The actual logic of a promise
+// Should we call it Future?
+class PromiseBody[T]
```

Comment:
This is the second part of ControlPayload. It is not a promise so let's remove that word. I would suggest ControlCommand. 

Also can we merge these and the ControlMessage class?

Reply:
No, we cannot merge these with ControlMessage class because they are not the same thing right now. We don't have proper handlers for ControlMessages so I think we have to migrate the handling logic using the promise handlers.